src/main/java/services/RestoreFactory.java

```diff
@@ -391,6 +393,17 @@ private InputStream getRestoreXmlHelper(String restoreUrl, HqAuth auth) {
         log.info("Restoring at domain: " + domain + " with auth: " + auth + " with url: " + restoreUrl);
         HttpHeaders headers = auth.getAuthHeaders();
         headers.add("x-openrosa-version", "2.0");
+
+        if (getCaseId() != null) {
+            // Add MAC header for case restores
+            try {
+                headers.add("X-MAC-DIGEST", RequestUtils.getHmac(formplayerAuthKey, restoreUrl));
```

Comment:
Rather than using the full URL can we just use the path? I think this will be better since formplayer may perform requests to Django via a different host than what Django is configured with.
```
digest = RequestUtils.getHmac(formplayerAuthKey, restorePath)
fullUrl = new StringBuilder(host).append(restorePath).toString()
```